### PR TITLE
Fixed Missing Source Event bug

### PR DESCRIPTION
**Pull request by HannesHil**
Fixes https://github.com/inovex/CalendarSync/issues/68

### CODE REVIEW

File: internal/sync/controller.go [hunk 1]
```diff
@@ -177,8 +177,13 @@ func (p Controller) diffEvents(sourceEvents []models.Event, sinkEvents []models.
 
 		switch {
 		case !exists:
+			if event.Metadata.SourceID != "p.sink.GetSinkID()" {
```
Comment (MichaelEischer) [inline internal/sync/controller.go:180]: The quotes are wrong. Currently that condition is always true.

Shouldn't the check be `event.Metadata.SourceID == p.sink.GetSinkID()`? Syncing eventA (from calendar A) yields eventACopy (calendar B, SourceID = calendar A). After deleting eventA, `case !exists` applies when processing eventACopy, and `p.sink` would be calendar A.

Comment (HannesHil) [reply internal/sync/controller.go:180]: yes true but at least currently there is no p.sink.GetSinkID() that's why i marked it as string.
For the actual bug please take a look at the linked issue :)

**Comment by MichaelEischer** [comment]
I've replaced the placeholder commit with a proper implementation.

**Review by alxndr13 — COMMENTED** [review]
lgtm, i'll add a `.mockery.yaml` file as that is the new preferred way to go now. (is in: https://github.com/inovex/CalendarSync/pull/81)

imo, we should rename the "GetSourceID" func in the Sink and Source as it does not match the Sink Adapter. We may strive for something more generic, like "GetAdapterID" or something like this. 

what do you think? @MichaelEischer

**Comment by MichaelEischer** [comment]
> We may strive for something more generic, like "GetAdapterID" or something like this.

I'd somewhat prefer `GetCalendarID`. That works for both source and sink, but is clear on what the ID identifies.

File: internal/sync/controller_test.go [hunk 2]
```diff
@@ -306,6 +310,41 @@ func (suite *ControllerTestSuite) TestDeleteEventsNotInSink() {
 	suite.sink.AssertNumberOfCalls(suite.T(), "DeleteEvent", expectedDelete)
 }
 
+// TestDeleteEventsNotInSink verifies that if events are present in the sink-adapter, but not in the source, these
```
[the remark this answers is not in the document]
Comment (MichaelEischer) [reply internal/sync/controller_test.go]: I've updated the description.

**Comment by alxndr13** [comment]
> > We may strive for something more generic, like "GetAdapterID" or something like this.
> 
> 
> 
> I'd somewhat prefer `GetCalendarID`. That works for both source and sink, but is clear on what the ID identifies.

perfect. will create something once i'm home.

**Comment by MichaelEischer** [comment]
> perfect. will create something once i'm home.

@alxndr13 I've already added a commit to apply that rename.

**Comment by alxndr13** [comment]
> > perfect. will create something once i'm home.
> 
> @alxndr13 I've already added a commit to apply that rename.

whoops, did not saw that on the Github iOS App. lgtm

**Review by alxndr13 — APPROVED** [review]

**Review by MichaelEischer — APPROVED** [review]

**Pull request merged**